packages/core/src/RenderingEngine/VideoViewport.ts

```diff
@@ -0,0 +1,509 @@
+    const videoElement = this.videoElement;
+    const renderFrame = this.renderFrame;
+
+    videoElement.currentTime = videoElement.duration;
+
+    if (videoElement.paused) {
+      // Need to wait for seek update
+      const seekEventListener = (evt) => {
+        renderFrame();
+
+        videoElement.removeEventListener('seeked', seekEventListener);
+      };
+
+      videoElement.addEventListener('seeked', seekEventListener);
+    }
+  }
+
+  public async setTime(timeInSeconds: number) {
+    const videoElement = this.videoElement;
+    const renderFrame = this.renderFrame;
+
+    videoElement.currentTime = timeInSeconds;
+
+    if (videoElement.paused) {
+      // Need to wait for seek update
+      const seekEventListener = (evt) => {
+        renderFrame();
+
+        videoElement.removeEventListener('seeked', seekEventListener);
+      };
+
+      videoElement.addEventListener('seeked', seekEventListener);
+    }
+  }
+
+  // Sets the frame number - note according to DICOM, this is 1 based
+  public async setFrame(frame: number) {
+    this.setTime((frame - 1) / this.fps);
+  }
+
+  public setProperties(videoInterface: VideoViewportProperties) {
+    if (videoInterface.loop !== undefined) {
+      this.videoElement.loop = videoInterface.loop;
+    }
+
+    if (videoInterface.muted !== undefined) {
+      this.videoElement.muted = videoInterface.muted;
+    }
+
+    if (videoInterface.playbackRate !== undefined) {
+      this.setPlaybackRate(videoInterface.playbackRate);
+    }
+  }
+
+  public setPlaybackRate(rate = 1) {
+    // Minimum playback speed in chrome is 0.0625 compared to normal
+    if (rate < 0.0625) {
+      this.pause();
+      return;
+    }
+    if (!this.videoElement) {
+      return;
+    }
+    this.videoElement.playbackRate = rate;
+    this.play();
+  }
+
+  public setScrollSpeed(scrollSpeed = 1, unit = 'f') {
+    this.scrollSpeed = unit === 's' ? scrollSpeed * this.fps : scrollSpeed;
+  }
+
+  public getProperties = (): VideoViewportProperties => {
+    return {
+      loop: this.videoElement.loop,
+      muted: this.videoElement.muted,
+    };
+  };
+
+  public resetProperties() {
+    this.setProperties({
+      loop: false,
+      muted: true,
+    });
+  }
+
+  public setCamera(
+    videoInterface: VideoViewportProperties // TODO use a different interface here.
+  ): void {
+    if (videoInterface.pan !== undefined) {
+      this.videoCamera.pan = videoInterface.pan;
+    }
+
+    if (videoInterface.parallelScale !== undefined) {
+      this.videoCamera.parallelScale = videoInterface.parallelScale;
+    }
+
+    this.canvasContext.fillRect(0, 0, this.canvas.width, this.canvas.height);
+
+    if (this.isPlaying === false) {
+      // If its not replaying, just re-render the frame on move.
+      this.renderFrame();
+    }
+  }
+
+  public getCamera(): VideoViewportProperties {
+    return {
+      pan: this.videoCamera.pan,
+      parallelScale: this.videoCamera.parallelScale,
+    };
+  }
```

Comment:
I have significant concerns regarding annotation and segmentation tools. It would be ideal to calculate the ICamera and export it here, as this would ensure the proper functioning of our annotation/segmentation tools once we implement getImageData. Otherwise, we would need to extensively add custom logic for the Video in the tools, which would be quite challenging. Therefore, I strongly prefer to handle it in this manner.

The reason why our GPU and CPU StackViewport work seamlessly with the tools is because we simulate the computation of all ICameras for the CPU route (even though vtk Camera is not used). By enforcing this computation, we eliminate any explicit mention of the CPU vs GPU within the tools. See `getCameraCPU`